Add local-ai community container with Vulkan support

docjyJ opened the pull request:
A lot of GPU supported: https://vulkan.gpuinfo.org/listdevices.php?platform=linux

Image: https://github.com/docjyJ/aio-local-ai-vulkan

szaimen commented:
@docjyJ thanks a lot for this PR! :)

docjyJ commented:
Not yet, it's quite complicated to have Vulkan on my server. I have a very old version of the Linux kernel so I need to find a solution to update.

docjyJ commented:
Local AI doesn't have a Vulkan image for arm64...
Since my cloud is on arm, I'm stuck...

I've opened an issue: https://github.com/mudler/LocalAI/issues/5778

docjyJ commented:
I have Vulkan on my PC and am running a local instance of Nextcloud over HTTP, so I opted for simplicity with automatic configuration and explanations for accessing the web interface.

szaimen commented:
Thanks @docjyJ for continuing the work on this! 😊

I mean you could also try to build the image from source instead of proxying it. Then you could also build for arm64... However of course not sure how feasible this is...

szaimen commented on community-containers/local-ai-vulkan/readme.md:
````diff
@@ -0,0 +1,34 @@
+```Cadyfile
+http://local-ai.your-nc-domain.com {
+    # Local AI web interface haven't any authentication, so you should protect it
+    basic_auth {
+        # Username "Bob", password "hiccup"
+        Bob $2a$14$Zkx19XLiW6VYouLHR5NmfOFU0z2GTNmpkT/5qqR7hx4IjWJPDhjvG
````
Great solution! 😊

Should we maybe directly build support for this into https://github.com/szaimen/aio-caddy?

docjyJ replied:
I don't know, as the password is hard-coded, I think it's possible but we need to consider another solution than basic auth.
maybe this https://caddyserver.com/docs/caddyfile/directives/forward_auth

szaimen commented on community-containers/local-ai-vulkan/readme.md:
```diff
@@ -0,0 +1,34 @@
+> ARM devices are not supported by this container. It is only for x86_64 devices. (See: https://github.com/mudler/LocalAI/issues/5778)
+
+> [!CAUTION]
+> If the configuration is not correct, it should still work but it will use the CPU only and will not benefit from hardware acceleration.
```
Ah nice! I didnt know. I mean if we figure out the arm64 support, we could even make it the defwult local-ai container instead of a variant from it. Also see https://github.com/szaimen/aio-local-ai/issues/66

docjyJ replied:
Yes, I made some text with the image. If I remove /dev/dri, it only uses the CPU.

docjyJ replied:
without /dev/dri
![image](https://github.com/user-attachments/assets/09d6a37f-b631-4232-8d7a-c4a96a5aaa34)

with /dev/dri
![image](https://github.com/user-attachments/assets/b6de3d85-eb03-4abb-aa42-95565ebe1fc8)

In both cases it works but with /dev/dri it's really faster

docjyJ replied:
But cpu support arm :(

docjyJ commented:
![image](https://github.com/user-attachments/assets/1beee524-d0fb-4db2-a272-613443d05e26)
It's using my GPU. But I don't know why it's so slow...

szaimen commented on community-containers/local-ai-vulkan/readme.md:
```diff
@@ -0,0 +1,34 @@
+## Local AI
+This container bundles Local AI and auto-configures it for you.
+
+### Notes
```
I fear it only supports one afaik

docjyJ commented:
See that https://www.reddit.com/r/LocalLLaMA/comments/1j1swtj/vulkan_is_getting_really_close_now_lets_ditch/

docjyJ commented:
Yes, I think it could be the main container.

ARM support has an open PR: https://github.com/mudler/LocalAI/pull/5780

szaimen commented:
> Yes, I think it could be the main container.

Cool!

> ARM support has an open PR: [mudler/LocalAI#5780](https://github.com/mudler/LocalAI/pull/5780)

I see... Honestly, I would like to wait a few weeks and see if arm64 support evolves upstream so that we can push this forward here...

docjyJ commented:
I added web ui access with basic auth, we can support caddy community container.
Updated doc and use local AI image without preconfigured models (to save space disk, user can install models from local AI web ui)

docjyJ commented:
I saw that PR has had the Roadmap label for two weeks. Waiting for news.

docjyJ commented:
Yes, waiting for the next release.

szaimen reviewed (commented):
Some comments and then is this good to go from my POV :)
